pkg/cmd/rollbackcopy/etcdclientutils.go

```diff
@@ -0,0 +1,119 @@
+package rollbackcopy
+
+import (
+	"context"
+	"fmt"
+	"google.golang.org/grpc"
+	"io"
+	"k8s.io/klog"
+	"log"
+	"os"
+	"strings"
+	"time"
+
+	"go.etcd.io/etcd/clientv3"
+	"go.etcd.io/etcd/pkg/transport"
+)
+
+func getEtcdClient(endpoints []string) (*clientv3.Client, error) {
+	dialOptions := []grpc.DialOption{
+		grpc.WithBlock(), // block until the underlying connection is up
+	}
+
+	tlsInfo := transport.TLSInfo{
+		CertFile:      os.Getenv("ETCDCTL_CERT"),
+		KeyFile:       os.Getenv("ETCDCTL_KEY"),
+		TrustedCAFile: os.Getenv("ETCDCTL_CACERT"),
+	}
+	tlsConfig, err := tlsInfo.ClientConfig()
+
+	cfg := &clientv3.Config{
+		DialOptions: dialOptions,
+		Endpoints:   endpoints,
+		DialTimeout: 2 * time.Second,
+		TLS:         tlsConfig,
+	}
+
+	cli, err := clientv3.New(*cfg)
+	if err != nil {
+		return nil, fmt.Errorf("failed to make etcd client for endpoints %v: %w", endpoints, err)
+	}
+	return cli, err
+}
+
+func IsLeader(name string) (bool, error) {
+	cli, err := getEtcdClient(strings.Split(os.Getenv("ETCDCTL_ENDPOINTS"), ","))
+	if err != nil {
+		return false, err
+	}
+	defer cli.Close()
+
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
```

Comment:
There is no diff as far as I know. Here is a section from a book on this topic:
https://www.programming-books.io/essential/go/context-todo-vs-context-background-d5224e27ff724a33a79cb4e03a5eb333

Since getcli package is also using context.Background(), I used it here too.

Reply:
The stdlib source can show us that Background and TODO will be equivalent here:

https://golang.org/src/context/context.go?s=7426:7451#L198

I vote to use `context.Background` or `context.TODO` alone without the `WithCancel` wrapper. Why set up a deferred function call that does nothing? It only adds confusion.